### ISSUE TITLE
xterm (no such file or directory)

### ISSUE THREAD

**Issue by akhilleusuggo** [Botspot/pi-apps#503]
Trying to execute the app ''All is Well" , getting error `failed to execute child process xterm (no such file or directory)`

I'm not sure why , but maybe because I have installed the app Powerline-Shell , and does disable the default one .

Quick google , recommend installing xterm , but I don't remember have uninstalled it .

**Comment by Botspot** [comment]
This is likely due to your desktop environment trying to use a nonexistent terminal.

We could try to fix it, but I don't know how to do that. Instead, we'll work around it.  
Try running this to change the menu button:
```
DIRECTORY=~/pi-apps
echo "[Desktop Entry]
Name=All Is Well
GenericName=Update Helper and Fixer
Comment= ALL IS WELL is a bash script that allows users to easily update and upgrade their repositories and packages on linux. It will also fix any broken packages and dependencies.
Exec=$DIRECTORY/etc/terminal-run $HOME/All-is-well/aiw/aiwrpi.sh 'All Is Well'
Icon=$HOME/pi-apps/apps/All Is Well/icon-64.png
Terminal=false
StartupNotify=true
Type=Application
Categories=Utility;" > ~/.local/share/applications/aiw.desktop
```
So run that ^ , and then try running All is Well from the menu. See if it works now.

**Comment by akhilleusuggo** [comment]
> DIRECTORY=~/pi-apps
> echo "[Desktop Entry]
> Name=All Is Well
> GenericName=Update Helper and Fixer
> Comment= ALL IS WELL is a bash script that allows users to easily update and upgrade their repositories and packages on linux. It will also fix any broken packages and dependencies.
> Exec=$DIRECTORY/etc/terminal-run $HOME/All-is-well/aiw/aiwrpi.sh 'All Is Well'
> Icon=$HOME/pi-apps/apps/All Is Well/icon-64.png
> Terminal=false
> StartupNotify=true
> Type=Application
> Categories=Utility;" > ~/.local/share/applications/aiw.desktop

It works .

I think the Pi-apps need to take in count new implementation since it's going to break others .
maybe for "All Is Well" need to check what terminal is been used , then execute it from there . https://askubuntu.com/questions/640096/how-do-i-check-which-terminal-i-am-using

**Comment by akhilleusuggo** [comment]
> DIRECTORY=~/pi-apps
> echo "[Desktop Entry]
> Name=All Is Well
> GenericName=Update Helper and Fixer
> Comment= ALL IS WELL is a bash script that allows users to easily update and upgrade their repositories and packages on linux. It will also fix any broken packages and dependencies.
> Exec=$DIRECTORY/etc/terminal-run $HOME/All-is-well/aiw/aiwrpi.sh 'All Is Well'
> Icon=$HOME/pi-apps/apps/All Is Well/icon-64.png
> Terminal=false
> StartupNotify=true
> Type=Application
> Categories=Utility;" > ~/.local/share/applications/aiw.desktop

It works .

I think the Pi-apps need to take in count new implementation since it's going to break others .
maybe for "All Is Well" need to check what terminal is been used , then execute it from there . https://askubuntu.com/questions/640096/how-do-i-check-which-terminal-i-am-using

**Comment by Botspot** [comment]
Changes merged with commit d898407b036115312fbd10aef6598ed6a00d7810.

**Comment by akhilleusuggo** [comment]
thank you :)
